app/src/main/java/org/groundplatform/android/ui/datacollection/DataCollectionViewModel.kt

```diff
@@ -197,23 +197,24 @@ internal constructor(
   fun getTaskViewModel(taskId: String): AbstractTaskViewModel? {
     val viewModels = taskViewModels.value
 
-    val task = tasks.first { it.id == taskId }
-
     if (viewModels.containsKey(taskId)) {
       return viewModels[taskId]
     }
 
-    return try {
-      val viewModel = viewModelFactory.create(getViewModelClass(task.type))
-      taskViewModels.value[task.id] = viewModel
-
-      val taskData: TaskData? = if (shouldLoadFromDraft) getValueFromDraft(task) else null
-      viewModel.initialize(job, task, taskData)
-      viewModel
-    } catch (e: Exception) {
-      Timber.e("ignoring task with invalid type: $task.type")
-      null
-    }
+    return tasks
+      .firstOrNull { it.id == taskId }
```

Comment:
We can simplify the nesting by returning early. Then the rest of the code can remain flattened.

```
val task = tasks.firstOrNull { it.id == taskId } ?: return null
...
```

Reply:
the number of return statement is increaseing hence did this.

Reply:
Is that causing any lint errors?

Reply:
Yes